dff/messengers/discord/interface.py

```diff
@@ -0,0 +1,63 @@
+from io import BytesIO
+from logging import getLogger
+
+from dff.messengers.common.modules import discord
+from pydantic import HttpUrl
+
+from dff.messengers.common import CallbackMessengerInterface
+from dff.pipeline.types import PipelineRunnerFunction
+from dff.script.core.message import Attachments, Audio, Document, Image, Message, Video
+
+logger = getLogger(__name__)
+
+
+def extract_message_from_discord(message: discord.Message) -> Message:  # pragma: no cover
+    inn_mess = Message()
+    inn_mess.text = message.content
+
+    files = list()
+    if len(message.attachments) > 0:
+        first_attachment = message.attachments[0]
+        if first_attachment.content_type is not None:
+            content_type = first_attachment.content_type.split("/")[0]
+            if content_type == "audio":
+                files = [Audio(source=HttpUrl(first_attachment.url))]
+            elif content_type == "video":
+                files = [Video(source=HttpUrl(first_attachment.url))]
+            elif content_type == "image":
+                files = [Image(source=HttpUrl(first_attachment.url))]
+            elif content_type in ("application", "text"):
+                files = [Document(source=HttpUrl(first_attachment.url))]
+
+    inn_mess.attachments = Attachments(files=files)
+    return inn_mess
+
+
+async def cast_message_to_discord_and_send(channel: discord.abc.Messageable, message: Message) -> None:  # pragma: no cover
+    files = list()
+    if message.attachments is not None:
+        for file in message.attachments.files[:10]:
+            if file.source is not None:
+                files += [discord.File(BytesIO(file.get_bytes()), file.title)]
+
+    await channel.send(message.text, files=files)
```

Comment:
What happens if `message.text` is None?

Reply:
It should be fine, since first argument of `channel.send` is `Optional[str]`